pkg/api/file.go

```diff
@@ -198,6 +200,143 @@ func (s *server) fileUploadHandler(w http.ResponseWriter, r *http.Request) {
 	})
 }
 
+// FileUploadInfo contains the data for a file to be uploaded
+type FileUploadInfo struct {
+	name        string // file name
+	size        int64  // file size
+	contentType string
+	toEncrypt   bool
+	reader      io.Reader
+}
+
+// GetFileHTTPInfo extracts data for a file to be uploaded from an HTTP request
+// this function was extracted from `fileUploadHandler` and should eventually replace its current code, along with `storeFile`
```

Comment:
Why don't you just do this right now? I understand that would make the changeset bigger, but right now there is a lot of duplication. Also, if the stuff that you wrote here is wrong, it won't be caught by unit tests. Let's be bold and just do the refactoring.

Reply:
per @zbiljic's suggestion, i've removed this function.

refactoring should be done through #462 to keep this PR as small as possible